storybookjs/storybook
# Fix angular cli style loading with latest `@angular-devkit/build-angular`

Issue: https://github.com/storybooks/storybook/issues/4378 https://github.com/storybooks/storybook/issues/5239

## What I did
Add `buildOptions.sourceMap` and `buildOptions.optimization` by default to angular webpack config. So that getting angular config from `@angular-devkit/build-angular` does not fail.

Commit with breaking changes in angular is [here](https://github.com/angular/angular-cli/commit/4f8a5b7a55316f773c1ae80505864212dc278b6a#diff-ef298d20cc5b49a2407cabad6876a7b4R23)

## How to test

- In `angular-cli` example app, with `@angular-devkit/build-angular@v13` you will see all pages with `CSS not loaded properly` red message. Navigate to `Custom/styleUrls` tab and content does not show. With changes you will see content working correctly with 3 texts with different colours.
- Added unit test.

- Is this testable with Jest or Chromatic screenshots?
No
- Does this need a new example in the kitchen sink apps?
No
- Does this need an update to the documentation?
No
